core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -197,7 +197,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     protected abstract SuggestionBuilder<T> innerFromXContent(QueryParseContext parseContext, String name) throws IOException;
 
-    protected abstract SuggestionContext build(QueryShardContext context) throws IOException;
+    public SuggestionContext build(QueryShardContext context) throws IOException {
+        SuggestionContext suggestionContext = innerBuild(context);
+        // copy over common settings to each suggestion builder
+        populateCommonFields(context.getMapperService(), suggestionContext);
```

Comment:
@areek good point.  A simple solution would be to have a separate `newInstance` abstract method that each subclass implements, in order to just get the instance of the context, then populate the common fields on it, then call `innerBuild` on it.  Thoughts?